### DISCUSSION TITLE
Teensy 4.0 Support?

### DISCUSSION THREAD

**Discussion by SturdyFool10** [PlummersSoftwareLLC/NightDriverStrip#31] [Q&A]
Hello, I’m curious as to whether or not this supports a Teensy 4.0, as their smaller form factor is extremely helpful sometimes, not to mention their overclocking to over 1.1GHz

**Comment by Karbonforms** [comment]
Straight FastLed on Teensy (I use 4.1) is broken. I had to use OctoWS2811 (with a wrapper class for FastLed).

So you'd need a lib_dep for OctoWS2811, it's required memory allocations and a conditionally compiled CPixelLEDController derivative and FastLed constructor using said class.

**Comment by robertlipe** [comment]
I know it's been a long time, but I'm going to be the one to take the bullet and say "no" and ask that you mark this closed, please.

Teensy is a nifty SOC. In the time since this question was opened, the ESP32 ecosystem got really strong. The Arduino/ESP32 marriage is currently in counseling (see the header of https://github.com/pioarduino/platform-espressif32) but for better and worse, this project is currently pretty "in" on the Arduino and ESP32 intersection. 

I think I suggested once moving something to a less Arduino-ish way of doing something and asked if the project was married to the Arduino approach and Dave said it was married, had kids, and was staying together until the kids graduated.

There are definitely pieces of this project that make sense on other SOCs, but as a non-authoritiative answer, I think it's safe to say the focus of THIS project remains on the Arduino/ESP32 ecosystem - and honestly, is struggling to remain current on development even within this niche.

Can I imagine parts of this code rockin' on something like a triple-core BL808 or Sophgo SG2002 scorching the track at a >1ghz and, more importantly, with "no excuses" S/DRAM instead of paging serial PSRAM? You bet! But for MOST people and, I think, for the target audience of this project, the value proposition of the ESP32 with 1-2 core at 160-240Mhz but including WIFI and the wide (but not necessarily quality, #sorrynotsorry) arduino ecosystem means that this project's official code base is likely to remain focused on ESP32.

If one of us that cared wanted to port it in a way that didn't wreck the ESP32 path AND wanted to maintain it, maybe we could argue it in. Honestly, it'd probably be easier to do as a fork.

So if you're willing to accept "no, it does not currently support Teensy" as a correct answer, please consider marking this as correct. If you want to consider what it'd take to make it a reality - and are willing to help with the work - buzz me.
